### ISSUE TITLE
Deprecated API usage in DE

### ISSUE THREAD

**Issue by lslachciak** [Samsung/TAU-Design-Editor#66]
DE uses deprecated document.registerElement API in many places
```
design-editor/src/pane/select-layer/layout-detail-element.js:const LayoutDetailElement = document.registerElement('closet-element-layout-detail', LayoutDetail);
design-editor/src/pane/select-layer/select-layer-element.js:const SelectLayerElement = document.registerElement('closet-select-layer', SelectLayer);
design-editor/src/pane/select-layer/alternative-selector-element.js:const AlternativeSelectorElement = document.registerElement('closet-alternative-selector', AlternativeSelector);
design-editor/src/pane/select-layer/info-element.js:const InfoElement = document.registerElement('info-bar', Info);
design-editor/src/pane/section-controller-element.js:const SectionControllerElement = document.registerElement('closet-section-controller', SectionController);
design-editor/src/pane/expandable-element.js:const ExpandableElement = document.registerElement('closet-expandable-element', Expandable);
design-editor/src/pane/popup-element.js:const PopupElement = document.registerElement('closet-popup-element', Popup);
design-editor/src/pane/grid-element.js:const GridElement = document.registerElement('closet-design-editor-grid', {
design-editor/src/pane/design-editor-element.js:const DesignEditorElement = document.registerElement('closet-design-editor', DesignEditor);
design-editor/src/pane/ruler-element.js:const RulerElement = document.registerElement('closet-design-editor-ruler', {
design-editor/src/pane/snap-guide-element.js:const SnapGuideElement = document.registerElement('closet-design-snap-guide', SnapGuide);
design-editor/src/panel/assistant/assistant-wizard.js:const AssistantWizardElement = document.registerElement('closet-assistant-wizard', AssistantWizard);
design-editor/src/panel/assistant/assistant-view-element.js:const AssistantViewElement = document.registerElement('closet-assistant-view', AssistantView);
design-editor/src/panel/toolbar-element.js:const ToolbarElement = document.registerElement('closet-toolbar-container-element', Toolbar);
design-editor/src/panel/animation/animation-edit-element.js:const AnimationEditElement = document.registerElement('closet-animation', AnimationEdit);
design-editor/src/panel/animation/animation-preview-element.js:const AnimationPreviewElement = document.registerElement('closet-animation-simple', AnimationPreview);
design-editor/src/panel/animation/animation-container-element.js:const AnimationContainerElement = document.registerElement('closet-animation-container', AnimationContainer);
design-editor/src/panel/panel.js:const PanelElement = document.registerElement('closet-panel', Panel);
design-editor/src/panel/configuration-design-area-element.js:const ConfigurationDesignAreaElement = document.registerElement('configuration-design-area', ConfigurationDesignArea);
design-editor/src/panel/property/attribute/attribute-element-styles.js:const AttributeStylesElement = document.registerElement('closet-attribute-styles', AttributeStyles);
design-editor/src/panel/property/attribute/attribute-element-checkbox.js:const AttributeCheckboxElement = document.registerElement('closet-checkbox-element', AttributeCheckbox);
design-editor/src/panel/property/attribute/attribute-element-sthings.js:const AttributeSmartThingsElement = document.registerElement('closet-attribute-element-smartthings', AttributeSmartThings);
design-editor/src/panel/property/attribute/attribute-element-layout.js:const AttributeLayoutElement = document.registerElement('closet-attribute-element-layout', AttributeLayout);
design-editor/src/panel/property/attribute/attribute-element-image.js:const AttributeImageElement = document.registerElement('closet-image-element', AttributeImage);
design-editor/src/panel/property/attribute/attribute-element.js:const AttributeElement = document.registerElement('closet-attribute-element', Attribute);
design-editor/src/panel/property/attribute/attribute-element-interactive.js:var AttributeInteractiveElement = document.registerElement('closet-attribute-interactive', AttributeInterative);
design-editor/src/panel/property/structure/structure-element.js:const StructureElement = document.registerElement('closet-structure-element', Structure);
design-editor/src/panel/property/property-container-element.js:const PropertyContainerElement = document.registerElement('closet-property-container-element', PropertyContainer);
design-editor/src/panel/property/behavior/behavior-element.js:const BehaviorElement = document.registerElement('closet-behavior-element', Behavior);
design-editor/src/panel/property/component/component-element.js:const ComponentElement = document.registerElement('closet-component-element', Component);
design-editor/src/panel/preview/interaction-view-toolbar-element.js:const InteractionViewToolbarElement = document.registerElement('closet-interaction-view-element-toolbar', InteractionViewToolbar);
design-editor/src/panel/preview/preview-element.js:const PreviewElement = document.registerElement('closet-preview-element', Preview);
design-editor/src/panel/preview/preview-toolbar-element.js:const PreviewToolbarElement = document.registerElement('closet-preview-element-toolbar', PreviewToolbar);
design-editor/src/panel/preview/interaction-view-element.js:const InteractionViewElement = document.registerElement('closet-interaction-view-element', InteractionView);
design-editor/src/panel/tooltip-element.js:const TooltipElement = document.registerElement('closet-tooltip', Tooltip);
design-editor/src/panel/wizards/project-wizard-element.js:const ProjectWizardElement = document.registerElement('closet-project-wizard', ProjectWizard);
design-editor/src/panel/wizards/components/grouped-button-element.js:const GruppedButtonElement = document.registerElement('closet-grouped-button', GruppedButton);
design-editor/src/panel/wizards/components/thumbnail-element.js:const ThumbnailElement = document.registerElement('closet-pw-thumbnail', Thumbnail);
design-editor/src/panel/wizards/components/text-editor-element.js:const TextEditorElement = document.registerElement('text-editor', TextEditor);
design-editor/src/panel/wizards/components/category-element.js:const CategoryElement = document.registerElement('closet-pw-category', Category);
design-editor/src/panel/wizards/components/file-picker-element.js:const FilePickerElement = document.registerElement('closet-file-picker', FilePicker);
design-editor/src/panel/wizards/page-wizard-element.js:const PageWizardElement = document.registerElement('closet-page-wizard', PageWizard);
design-editor/src/vsc/brackets-status-bar.js:const BracketsStatusBarElement = document.registerElement('brackets-status-bar', BracketsStatusBar);
design-editor/src/brackets/brackets-status-bar.js:const BracketsStatusBarElement = document.registerElement('brackets-status-bar', BracketsStatusBar);

```
This should be done before https://bugs.chromium.org/p/chromium/issues/detail?id=660759 is closed

**Comment by lslachciak** [comment]
currently warning message in chromium is 

document.registerElement is deprecated and **will be removed in M80, around February 2020.** Please use window.customElements.define instead. See https://www.chromestatus.com/features/4642138092470272 and https://developers.google.com/web/updates/2019/07/web-components-time-to-upgrade for more details.

this issue will affect first code.tizen.org and VSCode a little bit later

**Comment by lslachciak** [comment]
In order to simulate Febaruary browser behaviour I run fllowing command
**/opt/google/chrome/chrome --disable-blink-features=ShadowDOMV0,CustomElementsV0,HTMLImports**

result:
Live Preview for code.tizen.org examples is not working,

**Comment by korneliakobiela** [comment]
Here's what I found:

1. Custom elements v0 are deprecated since 2019 (since Chrome 70 release) and will be removed in Chrome 80 version
2. I forecast that support in VSCode will be removed in a couple of months.
3. Design Editor is built using Custom Elements and will stop working while the support will end.

We have some solutions to this situation:

1. Google recommends some polyfil for migration phase, but its not the best idea. This solution is quite old and I think it will have a bad influence on DE performance.

2. We have to think about rewrite some components of DE to allow it works in the future.

Here is helpful article: https://developers.google.com/web/updates/2019/07/web-components-time-to-upgrade

**Comment by korneliakobiela** [comment]
I think there is one more problem, named dress-element. This looks like some parent class and our custom implementation of custom elements. I will investigate its flow.

**Comment by korneliakobiela** [comment]
In Design Editor we have various types of Custom Elements:
1. Extends `DressElement`
2. Not extends `DressElement` but extends `HTMLElement`
3. Not extends `DressElement` but extends other HTML*Element e.g.: `HTMLCanvasElement`

In the third option, we need to do these steps:
1. Check if it is done using JS Class not a function constructor
2. Remove line with register element function and assignment to a variable.
3. Change export statement, we have to remove definition assignment, it's useless with the new API.
4. Change all the imports.

The example you can see in new PR: https://github.com/Samsung/TAU-Design-Editor/pull/469

**Comment by korneliakobiela** [comment]
@singa2000, @lslachciak I have a question. 
During my research, I've found some information which can be probably an issue. Do we need support Design Editor on Safari browser? If yes, this problem is bigger than we expected. Safari decided to not support Custom Elements extended Native HTML Elements and we will need more refactor on this field. Now we don't support any other browser besides Chromium-based ones.

> Note: Only Chrome 67 supports customized built-in elements (status) right now. Edge and Firefox will implement it, but Safari has chosen not to implement it. This is unfortunate for accessibility and progressive enhancement.  
Source: https://developers.google.com/web/fundamentals/web-components/customelements?fbclid=IwAR33GuBqOVR6JZBiCyAPWbKrzE8AxpBvMqyQD0KeUNuFx31xfZYSqG8znxY#extendhtm

**Comment by lslachciak** [comment]
We can have different code.tizen.org users.
As we discussed today morning there is an universal solution  way to inherit form general HTMLElement

**Comment by korneliakobiela** [comment]
Today I've fixed 3 new components. Tomorrow I think I will be able to fix New Page Wizard and show more instructions and more examples on how to fix the elements. Sometimes it is tricky to solve all our use-cases.

**Comment by korneliakobiela** [comment]
I have some problem with Dress Element - with the new API Event Listeners doesn't work and there is JQuery use. It doesn't cause exceptions on console and I don't know what the particular problem is. So I have to rewrite this module from scratch.

**Comment by korneliakobiela** [comment]
I've implemented a new parent class for DE Elements. It is named Component and should be an ancestor for our components (instead of Dress Element)

For now, it has two features:
- `bindEvents` - for binding events listeners ( similar to Dress Element function, ready to use with Custom Elements V0
- `createFromTemplate` - creates element from HTML template.

Except that the whole Page Wizard was rewritten to a proper style. It is a bigger thing that I thought Tomorrow I will publish instruction for the change process.

**Comment by korneliakobiela** [comment]
I've changed all the registerElement functions to define() ones. It is now working yet, but I've made some PR: https://github.com/Samsung/TAU-Design-Editor/pull/473

**Comment by korneliakobiela** [comment]
The issue for DEMO Mode is fixed.
